### PR TITLE
initial apis addes

### PR DESCRIPTION
**Pull request by au1994**
@vaidik
@vishesh92
@dash1291

### CODE REVIEW

**Comment by dash1291** [comment]
Always try to make your commit messages meaningful.

**Comment by vaidik** [comment]
- Please do not commit `pyc` file.
- Read about `.gitignore`.

**Comment by vishesh92** [comment]
Please fix all the code style issues.

File: outermysite/polls/imdbExactApi.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import requests
+import sys
+import traceback
+from bs4 import BeautifulSoup
+
+def getMovieResults(Id):
```
Comment (vaidik) [inline outermysite/polls/imdbExactApi.py:6]: Every language usually has some convention on how to name variables and functions. For example, Java uses camel casing. In python, underscores are preferred in python for functions and variable names. Class names use upper camel case.

File: outermysite/polls/imdbExactApi.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    
+    url = "http://www.imdb.com/title/"+Id+"/?ref_=fn_al_tt_1"
+    
+    try:
+        r = requests.get(url)
+    except requests.exceptions.RequestException as e:
+        print "Unable to connect"
```
Comment (vaidik) [inline outermysite/polls/imdbExactApi.py:13]: No point in printing messages in a web server. Where are you going to look?

Log them instead. Try https://docs.python.org/2/howto/logging.html

File: outermysite/polls/imdbExactApi.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        raise SystemExit("Slow internet connection")
```
Comment (vaidik) [inline outermysite/polls/imdbExactApi.py:14]: Really? When you are running on a server, will `slow internet connection` be an issue?

File: outermysite/polls/imdbExactApi.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    jsonArray = []
+    
+    if movieDetails==None:
+        print "No results found"
+        jsonObject={}
+        jsonObject['Error']="No results found"
+        return jsonObject
+    
+    description = movieDetails.find("div", {"class" : "summary_text"})
+
```
Comment (vaidik) [inline outermysite/polls/imdbExactApi.py:29]: Probably too much spacing if you are adding a blank line after every line of code.

**Comment by dash1291** [comment]
@au1994 I can see quite a few code style issues especially around naming. Please go through https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions and make those changes

File: outermysite/polls/imdbExactApi.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    releaseDate = titleDetails.find(
+            "meta",
+            {"itemprop" : "datePublished"}
+            )
+    
+    genre = titleDetails.find_all("span", {"itemprop" : "genre"})
+    
+    jsonObject = {}
+    
+    if not(title == None):
```
Comment (vaidik) [inline outermysite/polls/imdbExactApi.py:53]: When there is only one condition you want to do `not equal to` on. Then you should do: `title != None`

Comment (vaidik) [inline outermysite/polls/imdbExactApi.py:53]: Also `title is not None` is even better. Read this: http://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is-in-python

File: outermysite/polls/imdbExactApi.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    titleDetails = soup.find("div", {"class" : "title_wrapper"})
+    
+    title = titleDetails.find("h1", {"itemprop" : "name"})
+    
+    duration = titleDetails.find("time", {"itemprop" : "duration"})
+    
+    releaseDate = titleDetails.find(
+            "meta",
+            {"itemprop" : "datePublished"}
+            )
```
Comment (dash1291) [inline outermysite/polls/imdbExactApi.py:47]: Improper indentation here

File: outermysite/polls/imdbExactApi.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    if not(actor == None):
+        actors=""
+        for item in actor:
+            actors = actors + item.get_text().strip() + " "
+        jsonObject['Actors'] = actors
+    
+    if not(rating == None):
+        jsonObject['Rating'] = rating.get_text().strip()
+    
+    return jsonObject
```
Comment (vaidik) [inline outermysite/polls/imdbExactApi.py:92]: Camel casing again. Use underscores. `json_object` Fix this everywhere

File: outermysite/polls/imdbSearchApi.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import requests
+import sys
+import traceback
+from bs4 import BeautifulSoup
+
+def getSearchResults(movie):
+    
+    url = "http://www.imdb.com/find?s=tt&q="+movie
```
Comment (vaidik) [inline outermysite/polls/imdbSearchApi.py:8]: Space around operators.

File: outermysite/polls/imdbSearchApi.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    
+    try:
+        r = requests.get(url)
+    except requests.exceptions.RequestException as e:
+        raise SystemExit("Slow internet connection")
+    
+    soup = BeautifulSoup(r.content, "html.parser")
+    movies = soup.find_all("td", {"class":"result_text"})
```
Comment (vaidik) [inline outermysite/polls/imdbSearchApi.py:16]: Space after ':'

File: outermysite/polls/imdbExactApi.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
+import requests
+import sys
+import traceback
+from bs4 import BeautifulSoup
+
+def getMovieResults(Id):
+    
+    url = "http://www.imdb.com/title/"+Id+"/?ref_=fn_al_tt_1"
```
Comment (vishesh92) [inline outermysite/polls/imdbExactApi.py:8]: Space around operators. check for this everywhere.

File: outermysite/polls/imdbSearchApi.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    
+    jsonArray = [] 
+    
+    if len(movies)==0:
+        jsonObject={}
+        jsonObject['Error']="No results found"
```
Comment (vaidik) [inline outermysite/polls/imdbSearchApi.py:22]: Space around `=`

**Pull request merged**